### ISSUE TITLE
Haralick Features not calculating

### ISSUE THREAD

**Issue by saramcardle** [qupath/qupath#358]
In m4, Haralick features will not calculate. I've tried it on SLICs, tiles, cells, and annotations.  When going through the menu Analyze > Calculate Features > Add Intensity Features, it will select cells/detections as normal, but no measurement appears.  In a script, this error appears:

```
ERROR: Error running plugin: java.lang.IllegalArgumentException: No double parameter with key 'haralickMin'
    at java.base/java.util.concurrent.FutureTask.report(Unknown Source)
    at java.base/java.util.concurrent.FutureTask.get(Unknown Source)
    at qupath.lib.plugins.AbstractPluginRunner.awaitCompletion(AbstractPluginRunner.java:193)
    at qupath.lib.plugins.AbstractPluginRunner.runTasks(AbstractPluginRunner.java:157)
    at qupath.lib.gui.plugins.PluginRunnerFX.runTasks(PluginRunnerFX.java:94)
    at qupath.lib.plugins.AbstractPlugin.runPlugin(AbstractPlugin.java:156)
    at qupath.lib.algorithms.IntensityFeaturesPlugin.runPlugin(IntensityFeaturesPlugin.java:355)
    at qupath.lib.gui.scripting.QPEx.runPlugin(QPEx.java:260)
    at qupath.lib.gui.scripting.QPEx.runPlugin(QPEx.java:280)
    at qupath.lib.gui.scripting.QPEx$runPlugin.callStatic(Unknown Source)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:196)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:216)
    at Script1.run(Script1.groovy:3)
    at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:317)
    at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:155)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:766)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:696)
    at qupath.lib.gui.scripting.DefaultScriptEditor.executeScript(DefaultScriptEditor.java:676)
    at qupath.lib.gui.scripting.DefaultScriptEditor$2.run(DefaultScriptEditor.java:1033)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
    at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
    at java.base/java.lang.Thread.run(Unknown Source)
  Caused by No double parameter with key 'haralickMin'        at qupath.lib.plugins.parameters.ParameterList.getDoubleParameterValue(ParameterList.java:386)
        at qupath.lib.plugins.parameters.ParameterList.getDoubleParameterValue(ParameterList.java:427)
        at qupath.lib.algorithms.IntensityFeaturesPlugin$HaralickFeaturesComp.updateFeatures(IntensityFeaturesPlugin.java:1056)
        at qupath.lib.algorithms.IntensityFeaturesPlugin.processObject(IntensityFeaturesPlugin.java:628)
        at qupath.lib.algorithms.IntensityFeaturesPlugin$IntensityFeatureRunnable.run(IntensityFeaturesPlugin.java:454)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
INFO: Processing complete in 0.01 seconds
INFO: 
qupath.lib.algorithms.IntensityFeaturesPlugin  {"pixelSizeMicrons": 2.0,  "region": "ROI",  "tileSizeMicrons": 25.0,  "colorOD": true,  "colorStain1": true,  "colorStain2": true,  "colorStain3": false,  "colorRed": false,  "colorGreen": false,  "colorBlue": false,  "colorHue": false,  "colorSaturation": false,  "colorBrightness": false,  "doMean": false,  "doStdDev": false,  "doMinMax": false,  "doMedian": false,  "doHaralick": true,  "haralickDistance": 1,  "haralickBins": 32}
INFO: Result: true
```

**Comment by petebankhead** [comment]
Thanks, it looks like this broke for RGB images while a) fixing Haralick features for fluorescence images, and 2) making the way parameters are requested stricter (failing if the parameter is missing, rather than returning null).

It should be fixed by https://github.com/petebankhead/qupath/commit/685c4a6f29465457263ef86e0fe6cf62c4f942a2

**Comment by petebankhead** [comment]
This should be resolved in [v0.2.0-m5](https://github.com/qupath/qupath/releases/tag/v0.2.0-m5) - if not please feel free to reopen the issue.
